diff --git a/chrome/browser/ui/tabs/tab_strip_prefs.cc b/chrome/browser/ui/tabs/tab_strip_prefs.cc
index ca0afc6fd0..06e7ce0ee5 100644
--- a/chrome/browser/ui/tabs/tab_strip_prefs.cc
+++ b/chrome/browser/ui/tabs/tab_strip_prefs.cc
@@ -24,11 +24,11 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(prefs::kOrganizerPanelPinnedToTabstrip, true);
   registry->RegisterBooleanPref(prefs::kEverythingMenuPinnedToTabstrip, true);
   registry->RegisterBooleanPref(prefs::kTabScrollButtonsPinnedToTabstrip, true);
-  registry->RegisterBooleanPref(prefs::kVerticalTabsEnabled, false);
+  registry->RegisterBooleanPref(prefs::kVerticalTabsEnabled, true);
   registry->RegisterBooleanPref(
       prefs::kVerticalTabsExpandOnHoverEnabled,
       tabs::kVerticalTabsExpandOnHoverDefaultEnabled.Get());
-  registry->RegisterBooleanPref(prefs::kVerticalTabsEnabledFirstTime, false);
+  registry->RegisterBooleanPref(prefs::kVerticalTabsEnabledFirstTime, true);
   registry->RegisterBooleanPref(prefs::kVerticalTabsCollapsedState, false);
   registry->RegisterIntegerPref(prefs::kVerticalTabsUncollapsedWidth,
                                 kVerticalTabStripDefaultUncollapsedWidth);
